crates/re_viewer/src/ui/override_ui.rs

```diff
@@ -0,0 +1,266 @@
+            .filter_map(|vis| view_systems.get_by_identifier(*vis).ok())
+            .map(|vis| vis.visualizer_query_info().queried);
+
+        // Accmulate all the components if there are multiple visualizers
+        let mut components = components_per_visualizer.next().unwrap_or_default();
+        for mut rest in components_per_visualizer {
+            components.append(&mut rest);
+        }
+
+        // Empty space views of every available types
+        for component in components.difference(active_components) {
+            // If we don't have an override_path we can't set up an initial override
+            // this shouldn't happen if the `DataResult` is valid.
+            let Some(override_path) = data_result.override_path() else {
+                continue;
+            };
+
+            // If there is no registered editor, don't let the user create an override
+            if !ctx.component_ui_registry.has_registered_editor(component) {
+                continue;
+            }
+
+            if ui.button(component.as_str()).clicked() {
+                // We are creating a new override. We need to decide what initial value to give it.
+                // - First see if there's an existing splat in the recording.
+                // - Next see if visualizer system wants to provide a value.
+                // - Finally, fall back on the default value from the component registry.
+
+                let components = [*component];
+
+                let mut splat_cell: DataCell = [InstanceKey::SPLAT].into();
+                splat_cell.compute_size_bytes();
+
+                let Some(mut initial_data) = store
+                    .latest_at(query, &data_result.entity_path, *component, &components)
+                    .and_then(|result| result.2[0].clone())
+                    .and_then(|cell| {
+                        if cell.num_instances() == 1 {
+                            Some(cell)
+                        } else {
+                            None
+                        }
+                    })
+                    .or_else(|| {
+                        system_for_initial_value.initial_override_value(
+                            ctx,
+                            query,
+                            store,
+                            &data_result.entity_path,
+                            component,
+                        )
+                    })
+                    .or_else(|| {
+                        ctx.component_ui_registry.default_value(
+                            ctx,
+                            query,
+                            store,
+                            &data_result.entity_path,
+                            component,
+                        )
+                    })
+                else {
+                    re_log::warn!("Could not identify an initial value for: {}", component);
+                    return;
+                };
+
+                initial_data.compute_size_bytes();
```

Comment:
why is this needed?

Reply:
:shrug: it's always needed when we write data to the store or else we get errors about using data that hasn't been sized yet.